site/src/pages/WorkspacesPage/BatchActions.tsx

```diff
@@ -0,0 +1,142 @@
+import TextField from "@mui/material/TextField";
+import { Box } from "@mui/system";
+import { deleteWorkspace, startWorkspace, stopWorkspace } from "api/api";
+import { Workspace } from "api/typesGenerated";
+import { ConfirmDialog } from "components/Dialogs/ConfirmDialog/ConfirmDialog";
+import { displayError } from "components/GlobalSnackbar/utils";
+import { useState } from "react";
+import { useMutation } from "react-query";
+import { MONOSPACE_FONT_FAMILY } from "theme/constants";
+
+export const useBatchActions = (options: {
+  onSuccess: () => Promise<void>;
```

Comment:
I think this should be typed as `() => (void | Promise<void>)`, just so components are free to use sync or async logic when it makes sense for them

Reply:
This component has a very specific use case and should only be used in one place so I would not try to make its interface generic to be re-used